pkg/certs/certgen_test.go

```diff
@@ -47,9 +47,11 @@ func TestGenerateCerts(t *testing.T) {
 				t.Errorf("wanted no error, got error %q", err)
 			}
 
+			// If using a custom lifetime, validate the certs
+			// as of an hour before the intended expiration.
 			currentTime := time.Now()
 			if tc.config.Lifetime != 0 {
-				currentTime = time.Now().Add(24 * time.Duration(tc.config.Lifetime) * time.Hour).Add(-5 * time.Second)
+				currentTime = time.Now().Add(24 * time.Hour * time.Duration(tc.config.Lifetime)).Add(-time.Hour)
```

Comment:
```suggestion
				currentTime = currentTime.Add(24 * time.Hour * time.Duration(tc.config.Lifetime)).Add(-time.Hour)
```

nit: Not sure if this makes a real difference, at least it makes it so we don't fetch the now twice?

Reply:
doesn't hurt, done